third_party/nvfuser/csrc/ops/alias.cpp

```diff
@@ -371,4 +372,285 @@ TensorView* transpose(TensorView* x) {
   return transpose(x, 0, 1);
 }
 
+// Padding widths are assumed to be non-negative. Currently there's no
+// validation.
+TensorView* pad(TensorView* inp, const std::vector<Val*>& pad_widths) {
+  const auto& inp_dom = inp->domain()->noReductions();
+
+  const auto ndims = inp_dom.size();
+
+  TORCH_CHECK(
+      pad_widths.size() % 2 == 0 && pad_widths.size() / 2 <= ndims,
+      "Invalid number of padding widths: ",
+      pad_widths.size());
+
+  const auto num_padded_dims = pad_widths.size() / 2;
+  const auto num_non_padded_dims = ndims - num_padded_dims;
+
+  std::vector<IterDomain*> root_ids(ndims);
+  std::vector<IterDomain*> rfactor_ids(ndims);
+
+  // PadOp requires pad widths for all dimensions, even for non-padded
+  // ones.
```

Comment:
This doesn't seem true to me based on PadOp impl. Are you sure?

Reply:
For avoiding ambiguity, I decided to make the padding vector of `PadOp` always have the width vals for all dimensions, no matter if each dimension is actually padded. Added error checks to `PadOp` constructor and methods.